pam/internal/adapter/gdmmodel_test.go

```diff
@@ -0,0 +1,2343 @@
+package adapter
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"reflect"
+	"slices"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/msteinert/pam/v2"
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/authd"
+	"github.com/ubuntu/authd/internal/brokers/responses"
+	"github.com/ubuntu/authd/internal/log"
+	"github.com/ubuntu/authd/pam/internal/gdm"
+	"github.com/ubuntu/authd/pam/internal/pam_test"
+	"github.com/ubuntu/authd/pam/internal/proto"
+	pam_proto "github.com/ubuntu/authd/pam/internal/proto"
+	"golang.org/x/exp/constraints"
+)
+
+var gdmTestPrivateKey *rsa.PrivateKey
+var gdmTestSequentialMessages atomic.Int64
+
+const gdmTestIgnoredMessage string = "<ignored>"
+
+type gdmConvHandler struct {
+	mu *sync.Mutex
+	t  *testing.T
+
+	wantRequests        []gdm.RequestType
+	handledRequests     []gdm.RequestType
+	allRequestsReceived chan struct{}
+
+	receivedEvents    []*gdm.EventData
+	wantEvents        []gdm.EventType
+	allEventsReceived chan struct{}
+
+	pendingEvents        []*gdm.EventData
+	pendingEventsFlushed chan struct{}
+
+	supportedLayouts []*authd.UILayout
+	receivedBrokers  []*authd.ABResponse_BrokerInfo
+	selectedBrokerID string
+
+	currentStageCond    sync.Cond
+	currentStage        pam_proto.Stage
+	stageChanges        []pam_proto.Stage
+	lastNotifiedStage   *pam_proto.Stage
+	currentStageChanged chan struct{}
+}
```

Comment:
This is not test code, but helpers. I would move all of these definitions to a `helper_test.go` file to avoid cluttering the test file with all of these declarations.

Reply:
Fair, I was actually thinking on moving them too into different files but I was unsure about the naming, maybe they could be even `helper_gdmconvhandler_test.go` and `helper_gdmuimodel_test.go`?

Reply:
They could be moved into a single `helper_test.go` file, no need for more IMHO.

Reply:
added more, I think is fine since the two classes are separated and could technically be used for other kind of tests... But let me know.